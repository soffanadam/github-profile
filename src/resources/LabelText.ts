export enum LabelText {
  TITLE = 'Show Github Profile',
  INPUT_GITHUB_USERNAME = 'Input github username',
  CANNOT_CONNECT_TO_SERVER = 'Cannot connect to server'
}
