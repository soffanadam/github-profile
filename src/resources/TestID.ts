export enum TestID {
  SEARCH_FORM = 'search-form',
  SEARCH_FORM_INPUT = 'search-form-input'
}
