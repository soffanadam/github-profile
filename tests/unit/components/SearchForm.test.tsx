import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import {
  SearchForm,
  SearchFormProps,
  ERROR_CLASS
} from '@/components/SearchForm'
import { TestID } from '@/resources/TestID'
import { withSearchFormProps } from './utils'

const renderComponent = (props: SearchFormProps) => {
  return render(<SearchForm {...props} />)
}

describe('<SearchForm />', () => {
  it('renders the component', () => {
    const component = renderComponent(withSearchFormProps())
    expect(component).toBeTruthy()
    expect(component).toMatchSnapshot()
  })

  it('searches for text and submit', () => {
    const props = withSearchFormProps()
    const { getByTestId } = renderComponent(props)

    const input = getByTestId(TestID.SEARCH_FORM_INPUT)
    expect(input).toHaveValue('')
    fireEvent.change(input, {
      target: { value: 'welcome' }
    })
    expect(input).toHaveValue('welcome')

    fireEvent.submit(getByTestId(TestID.SEARCH_FORM))
    expect(props.onSubmit).toHaveBeenCalledTimes(1)
  })

  it('update update prefill', () => {
    const props = withSearchFormProps()
    const { getByTestId, rerender } = renderComponent(props)

    const input = getByTestId(TestID.SEARCH_FORM_INPUT)
    expect(input).toHaveValue('')

    props.prefill = 'aaaa'
    rerender(<SearchForm {...props} />)
    expect(input).toHaveValue(props.prefill)
  })

  it('indicate error', () => {
    const props = withSearchFormProps()
    props.error = 'error'
    const { getByTestId } = renderComponent(props)
    expect(getByTestId(TestID.SEARCH_FORM_INPUT)).toHaveClass(ERROR_CLASS)
  })

  it('indicate loading', () => {
    const props = withSearchFormProps()
    props.disabled = true
    const { getByTestId } = renderComponent(props)
    expect(getByTestId(TestID.SEARCH_FORM_INPUT)).toBeDisabled()
  })

  it('cannot submit empty value', () => {
    const props = withSearchFormProps()
    renderComponent(props)
    expect(props.onSubmit).toHaveBeenCalledTimes(0)
  })
})
