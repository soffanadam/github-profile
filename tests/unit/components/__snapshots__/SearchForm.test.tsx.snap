// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        data-testid="search-form"
      >
        <input
          class="py-2.5 px-5 w-full text-lg rounded shadow focus:shadow-lg focus:outline-none border border-transparent"
          data-testid="search-form-input"
          placeholder="Input github username"
          value=""
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      data-testid="search-form"
    >
      <input
        class="py-2.5 px-5 w-full text-lg rounded shadow focus:shadow-lg focus:outline-none border border-transparent"
        data-testid="search-form-input"
        placeholder="Input github username"
        value=""
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
